lib/services/messenger/translators/translator-scaffold.ts

```diff
@@ -47,30 +47,32 @@ export abstract class TranslatorScaffold implements Translator {
 	 * @param messages  Message to search.
 	 * @param format    Format used to encode the metadata.
 	 */
-	public static extractThreadId(
+	public static extractSource(
 		service: string,
 		messages: string[],
 		format?: MetadataEncoding,
-	): Promise<IdentifyThreadResponse> {
+	): Promise<SourceDescription> {
```

Comment:
Is this every likely to be asynchronous? If not, it probably shouldn't return a Promise.